### PR TITLE
fix: use tokio::time::Interval for scheduling anchor batches

### PR DESCRIPTION
**Pull request by nathanielc**
This leverages the time math of skipping missed ticks etc. We are only left with computing the time when we should start the first tick. For this we use chrono and then rely on Interval for the heavy lifting.

### CODE REVIEW

File: anchor-service/src/anchor_batch.rs [hunk 1]
```diff
@@ -76,28 +77,42 @@ impl AnchorService {
         pin_mut!(shutdown_signal);
 
         info!("anchor service started");
+        let period =
```
Comment (stbrody) [inline anchor-service/src/anchor_batch.rs:80]: nitpick, but could we pull all this logic for building the interval into a helper function to improve readability?

File: anchor-service/src/anchor_batch.rs [hunk 2]
```diff
@@ -76,28 +77,42 @@ impl AnchorService {
...
+            TimeDelta::from_std(self.anchor_interval).expect("anchor interval should be in range");
+        let buffer = period / 12;
```
Comment (stbrody) [inline anchor-service/src/anchor_batch.rs:82]: what is buffer doing?

**Review by stbrody — APPROVED** [review]

File: anchor-service/src/anchor_batch.rs (hunk 2 above)
Comment (nathanielc) [reply anchor-service/src/anchor_batch.rs:82]: its the buffer time we give ourselves to actually process the batch. I'll add a comment. For 1hour it ends up being 5 minutes.

**Review by smrz2001 — APPROVED** [review]
🚀

**Pull request merged**